model.py

```diff
@@ -6520,13 +6567,15 @@ def lookup_by_temporary_token(cls, _db, data_source, type, token):
         return credential
 
     @classmethod
-    def temporary_token_create(self, _db, data_source, type, patron, duration):
+    def temporary_token_create(
+            self, _db, data_source, type, patron, duration, value=None
+    ):
```

Comment:
The token we get from an OAuth provider is now stored as a temporary token. This means we need to make it possible to create a temporary token with a given value (which we got from the OAuth provider), rather than with a randomly generated value.